packages/tracing-internal/src/browser/metrics/index.ts

```diff
@@ -369,21 +369,27 @@ function _addPerformanceNavigationTiming(
 /** Create request and response related spans */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function _addRequest(transaction: Transaction, entry: Record<string, any>, timeOrigin: number): void {
-  _startChild(transaction, {
-    op: 'browser',
-    origin: 'auto.browser.browser.metrics',
-    description: 'request',
-    startTimestamp: timeOrigin + msToSec(entry.requestStart as number),
-    endTimestamp: timeOrigin + msToSec(entry.responseEnd as number),
-  });
+  // It is possible that we are collecting these metrics when the page hasn't finished loading yet, for example when the HTML slowly streams in.
+  // In this case, ie. when the document request hasn't finished yet, `entry.responseEnd` will be 0.
+  // In order not to produce faulty spans, where the end timestamp is before the start timestamp, we will only collect
+  // these spans when the responseEnd value is available. Relay would drop the entire transaction if it contained faulty spans.
+  if (entry.responseEnd !== 0) {
```

Comment:
```suggestion
  if (entry.responseEnd) {
```

save a few bytes and should be equally correct here?